Framework/Core/include/Framework/GroupedCombinations.h

```diff
@@ -0,0 +1,300 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef FRAMEWORK_PAIR_H
+#define FRAMEWORK_PAIR_H
+
+#include "Framework/ASoAHelpers.h"
+#include "Framework/GroupSlicer.h"
+#include "Framework/StringHelpers.h"
+
+#include <iostream>
+
+namespace o2::framework
+{
+
+// Define the type of a tuple interleaved from 2 tuples
+// Base struct without specialization
+template <typename T1, typename T2>
+struct InterleavedTupleType {
+  using interleaved_pack_t = typename interleave_pack<T1>::template with<T2>::type;
+  template <typename... Is>
+  struct TupleType {
+    using type = std::false_type;
+  };
+  template <std::size_t... Is>
+  struct TupleType<std::index_sequence<Is...>> {
+    using type = std::tuple<pack_element_t<Is, interleaved_pack_t>...>;
+  };
+  using type = typename TupleType<std::make_index_sequence<2>>::type;
+};
+
+// Proper specialization with tuples
+template <typename... T1s, typename... T2s>
+struct InterleavedTupleType<std::tuple<T1s...>, std::tuple<T2s...>> {
+  using interleaved_pack_t = typename interleave_pack<T1s...>::template with<T2s...>::type;
+  template <typename... Is>
+  struct TupleType {
+    using type = std::false_type;
+  };
+  template <std::size_t... Is>
+  struct TupleType<std::index_sequence<Is...>> {
+    using type = std::tuple<pack_element_t<Is, interleaved_pack_t>...>;
+  };
+  using type = typename TupleType<std::make_index_sequence<sizeof...(T1s) + sizeof...(T2s)>>::type;
+};
+
+// Create an instance of a tuple interleaved from given tuples
+template <typename... T1s, typename... T2s, std::size_t... Is>
+auto interleaveTuplesImpl(std::tuple<T1s...>& t1, std::tuple<T2s...>& t2, std::index_sequence<Is...>)
+{
+  return std::tuple_cat(std::make_tuple(std::get<Is>(t1), std::get<Is>(t2))...);
```

Comment:
Do you really need to use std::tuple? Can't you use our o2::framework::Pack and homogeneous_apply_ref?

Reply:
As far as I understand, a pack does not hold any data. But I create this interleaved tuple to output the collisions and tracks to the user. Like in usual combinations.

Reply:
Ok, let's keep this pending for now. If you address the `pack_to_tuple` comment maybe I get a clearer idea of  what you want to do here.